### ISSUE TITLE
Zero exit code when plugin not found

### ISSUE THREAD

**Issue by chinedufn** [rust-lang/mdBook#893]
## Problem

I started adding the `mdbook-linkcheck` plugin to my book build process and was confused when CI was still passing even though I had forgotten to install the plugin in my ci config.

```
#!/bin/bash -eo pipefail
(cd book && mdbook build) && cargo doc --no-deps --document-private-items -p foo && cp -R target/doc book/book/api
2019-03-25 16:25:00 [INFO] (mdbook::book): Book building has started
2019-03-25 16:25:00 [INFO] (mdbook::book): Running the html backend
2019-03-25 16:25:01 [INFO] (mdbook::book): Running the linkcheck backend
2019-03-25 16:25:01 [INFO] (mdbook::renderer): Invoking the "linkcheck" renderer
2019-03-25 16:25:01 [WARN] (mdbook::renderer): The command wasn't found, is the "linkcheck" backend installed?
2019-03-25 16:25:01 [WARN] (mdbook::renderer): 	Command: mdbook-linkcheck
 Documenting tw v0.1.11 (/root/foo-barg/app/crates/foo-bar-cli)
    Finished dev [unoptimized + debuginfo] target(s) in 2.16s

```

## Expected Behavior

Would expect that `mdbook build` would fail with a non zero exit code if we tried to use a plugin that it couldn't find.

**Comment by Dylan-DPC** [comment]
hi @chinedufn can you send a PR that makes the required change?

**Comment by segfaultsourcery** [comment]
This has been sitting idle for a while, and it's fairly low hanging fruit, so I fixed it.

**Comment by ehuss** [comment]
I think this is closed by #1122.
